evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -156,6 +156,21 @@ pub fn apply<DB: Database>(
             Ok(Bytes::new())
         }
         HEVMCalls::Accesses(inner) => Ok(accesses(state, inner.0)),
+        HEVMCalls::SetNonce(inner) => {
+            data.subroutine
+                .state()
+                .entry(inner.0)
+                .and_modify(|acc| acc.info.nonce = inner.1)
+                .or_insert_with(|| {
+                    let account_info = revm::AccountInfo { nonce: inner.1, ..Default::default() };
```

Comment:
This is incorrect - it is possible for the account not to be in `SubRoutine` but still exist, and this would overwrite any data attached to it.

We should also make it a hard requirement that the nonce we set is higher than or equal to the current nonce otherwise we will probably experience a lot of reports of tests not working correctly because people accidentally overwrite their contracts :smile: